Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -307,7 +310,8 @@ void GroupDetectors2::exec() {
            : 1.);
 
   // Build a new map
-  const size_t outIndex = formGroups(inputWS, outputWS, prog4Copy);
+  const size_t outIndex =
+      formGroups(inputWS, outputWS, prog4Copy, keepAll, unGroupedSet);
 
   // If we're keeping ungrouped spectra
   if (keepAll) {
```

Comment:
Can you check if the `moveOthers` call is overwriting the grouping defined by `IndexInfo`? I think it might.